Contents/WeaponModification/WeaponModificationPlayer.cs

```diff
@@ -62,38 +63,43 @@ internal class WeaponModificationPlayer : ModPlayer {
 					return;
 				}
 				Delay = globalItem.Delay;
+				shootspeed = 0;
+				knockback = 0;
+				damage = 1;
+				critChance = 0;
+				critDamage = 1;
+				List<Projectile> RegisterProjectile = new List<Projectile>();
 				for (int i = 1; i > 0; i--) {
 					if (currentIndex >= globalItem.ModWeaponSlotType.Length) {
 						currentIndex = 0;
 						break;
 					}
 					if (globalItem.ModWeaponSlotType[currentIndex] == -1) {
 						currentIndex++;
+						i++;
 						continue;
 					}
 					ModWeaponParticle modweapon = ModifierWeaponLoader.GetWeaponMod(globalItem.ModWeaponSlotType[currentIndex]);
 					shootspeed = modweapon.ShootSpeed;
-					knockback = modweapon.KnockBack;
 					modweapon.PreUpdate(Player);
 					modweapon.ModifyModificationDelay(Player, ref Delay, ref Recharge, ref i);
 					modweapon.ModifyAttack(Player, ref damage, ref knockback, ref shootspeed);
 					modweapon.ModifyCritAttack(Player, ref critChance, ref critDamage);
 					modweapon.PostUpdate(Player);
 					if (modweapon.ProjectileType != ProjectileID.None) {
 						for (int l = 0; l < modweapon.ShootAmount; l++) {
-							if (modweapon.Shoot(Player, Player.Center, (Main.MouseWorld - Player.Center).SafeNormalize(Vector2.Zero) * shootspeed, (int)(modweapon.ProjectileDamage * damage), knockback, i) != null) {
-								break;
-							}
-							Projectile proj = modweapon.Shoot(Player, l);
-							proj.knockBack = knockback;
-							if (Main.rand.Next(1, 101) < critChance)
-								damage *= 2 + critDamage;
-							proj.damage = (int)(modweapon.ProjectileDamage * damage);
-							proj.velocity = (Main.MouseWorld - proj.position).SafeNormalize(Vector2.Zero) * shootspeed;
+							RegisterProjectile.Add(modweapon.Shoot(Player, l));
 						}
 					}
 					currentIndex++;
 				}
+				foreach (Projectile proj in RegisterProjectile) {
+					proj.knockBack = knockback;
+					proj.damage = (int)(proj.damage * damage);
+					if (Main.rand.Next(1, 101) < critChance)
+						proj.damage = (int)(proj.damage * critDamage);
+					proj.velocity = (Main.MouseWorld - proj.position).SafeNormalize(Vector2.Zero) * shootspeed;
+				}
```

Comment:
It might be a good idea to also set `proj.netUpdate` to `true`, so it sends a packet so other players receive changes.